util.go

```diff
@@ -0,0 +1,253 @@
+package imgutil
+
+import (
+	"encoding/json"
+	"slices"
+	"strings"
+
+	v1 "github.com/google/go-containerregistry/pkg/v1"
+	"github.com/google/go-containerregistry/pkg/v1/empty"
+	"github.com/google/go-containerregistry/pkg/v1/mutate"
+	"github.com/google/go-containerregistry/pkg/v1/partial"
+	"github.com/google/go-containerregistry/pkg/v1/types"
+	"github.com/pkg/errors"
+)
+
+func GetConfigFile(image v1.Image) (*v1.ConfigFile, error) {
+	configFile, err := image.ConfigFile()
+	if err != nil {
+		return nil, err
+	}
+	if configFile == nil {
+		return nil, errors.New("missing config file")
+	}
+	return configFile, nil
+}
+
+func GetManifest(image v1.Image) (*v1.Manifest, error) {
+	manifest, err := image.Manifest()
+	if err != nil {
+		return nil, err
+	}
+	if manifest == nil {
+		return nil, errors.New("missing manifest")
+	}
+	return manifest, nil
+}
+
+func MutateManifest(i v1.Image, withFunc func(c *v1.Manifest) (mutateSubject, mutateAnnoations bool)) (v1.Image, error) {
+	// FIXME: put MutateManifest on the interface when `remote` and `layout` packages also support it.
```

Comment:
Each of remote, layout, and local should implement the v1.Image interface now. However the local manifest is going to contain garbage data like empty diffIDs and layer sizes of -1. We should probably detect where this is going to blow up and handle errors gracefully.